opentelemetry-exporter-geneva/geneva-uploader/src/ingestion_service/uploader.rs

```diff
@@ -0,0 +1,212 @@
+use chrono::{DateTime, Duration as ChronoDuration, Utc};
+
+use crate::config_service::client::{
+    GenevaConfigClient, GenevaConfigClientError, IngestionGatewayInfo,
+};
+use chrono::{Datelike, Timelike};
+use reqwest::{header, Client};
+use serde::Deserialize;
+use serde_json::Value;
+use std::collections::HashMap;
+use std::time::Duration;
+use thiserror::Error;
+use url::form_urlencoded::byte_serialize;
+use uuid::Uuid;
+
+/// Error types for the Geneva Uploader
+#[derive(Debug, Error)]
+pub(crate) enum GenevaUploaderError {
+    #[error("HTTP error: {0}")]
+    Http(#[from] reqwest::Error),
+    #[error("JSON error: {0}")]
+    Json(#[from] serde_json::Error),
+    #[error("Config service error: {0}")]
+    ConfigService(String),
+    #[allow(dead_code)]
```

Comment:
Remove this attribute?

Reply:
Sorry didn't get this, remove the `Others` variant - I have replaced it with `InternalError` to be consistent with the Config Client.

Reply:
I meant remove the `#[allow(dead_code)]` attribute. We are using `UploadFailed` variant in the code.